src/registerDynamicSymbol.c

```diff
@@ -0,0 +1,22 @@
+// RegisteringDynamic Symbols
+
+#include <R.h>
+#include <Rinternals.h>
+#include <R_ext/Rdynload.h>
+#include "has_topotoolbox.h"
+
+static const R_CMethodDef cMethods[] = {
+   {"has_topotoolbox", (DL_FUNC) &has_topotoolbox,1},
```

Comment:
I get a compiler warning

```
1. failed to assign RegisteredNativeSymbol for has_topotoolbox since has_topotoolbox is already defined in the 'topotoolboxr' namespace
```

which happens because we are defining `has_topotoolbox` on this line and our R function `has_topotoolbox` in `has_topotoolbox.R`. One way to fix this is described in [Writing R Extensions](https://cran.rstudio.com/doc/manuals/r-devel/R-exts.html#Registering-native-routines):

> Once routines are registered, they can be referred to as R objects if this is arranged in the useDynLib call in the package’s NAMESPACE file (see [useDynLib](https://cran.rstudio.com/doc/manuals/r-devel/R-exts.html#useDynLib)). So for example the stats package has

> Refer to all C/Fortran routines by their name prefixed by C_

    useDynLib(stats, .registration = TRUE, .fixes = "C_")

> in its NAMESPACE file, and then ansari.test’s default methods can contain

        pansari <- function(q, m, n)
            .C(C_pansari, as.integer(length(q)), p = as.double(q),
                as.integer(m), as.integer(n))$p

So I think we could do something like `useDynLib(topotoolboxr,.registration=TRUE,.fixes="C_")` and then

```
has_topotoolbox <- function(a){
    return(.C(C_has_topotoolbox,as.integer(a)))
}
```

to avoid this warning.

Reply:
So I added the '.fixes="C_"' the namespace file, and, as soon as I run "check()", it auto-delets the part. If i edit our has_topotoolbox.R to
`R
has_topotoolbox <- function(a){
    return(.C(C_has_topotoolbox,as.integer(a)))
}
`, then i can't run the check() function anymore.

Reply:
`bash
── Test failures ───────────────────────────────────────── testthat ────

> # This file is part of the standard setup for testthat.
> # It is recommended that you do not modify it.
> #
> # Where should you do additional test configuration?
> # Learn more about the roles of various files in:
> # * https://r-pkgs.org/testing-design.html#sec-tests-files-overview
> # * https://testthat.r-lib.org/articles/special-files.html
> 
> library(testthat)
> library(topotoolboxr)
Warning message:
failed to assign RegisteredNativeSymbol for has_topotoolbox to has_topotoolbox since has_topotoolbox is already defined in the 'topotoolboxr' namespace 
> 
> test_check("topotoolboxr")
[ FAIL 1 | WARN 0 | SKIP 0 | PASS 0 ]

══ Failed tests ════════════════════════════════════════════════════════════════
── Error ('test-has_topotoolbox.R:2:3'): has_topotoolbox equal 1? ──────────────
Error in `has_topotoolbox(1)`: object 'C_has_topotoolbox' not found
Backtrace:
    ▆
 1. ├─testthat::expect_equal(has_topotoolbox(1)[[1]], c(1)) at test-has_topotoolbox.R:2:3
 2. │ └─testthat::quasi_label(enquo(object), label, arg = "object")
 3. │   └─rlang::eval_bare(expr, quo_get_env(quo))
 4. └─topotoolboxr::has_topotoolbox(1)

[ FAIL 1 | WARN 0 | SKIP 0 | PASS 0 ]
Error: Test failures
Execution halted

1 error ✖ | 1 warning ✖ | 3 notes ✖
`

Reply:
Let me look into this. When I was playing around with it the other day, I didn't encounter this, but I wasn't running `devtools::check()` very often, so maybe there is some subtlety I missed.

Reply:
Oh, I also don't get the errors you were mentioning. Can you explain how you compile and run the project?

Reply:
Sorry, this turned out to be a combination of a misunderstanding I had of the R package compilation process and using an older version of roxygen2. 

You need to put the `.fixes = "C_"` part in `R/topotoolboxr-package.R` so that it is picked up by roxygen2 when it autogenerates `NAMESPACE`. I guess the `Generated by roxygen2: do not edit by hand` warning on `NAMESPACE` is serious.

The follow patch should work:

```diff
diff --git a/R/has_topotoolbox.R b/R/has_topotoolbox.R
index 22c43b6..3cc1944 100644
--- a/R/has_topotoolbox.R
+++ b/R/has_topotoolbox.R
@@ -4,7 +4,7 @@
 #'
 #' @export
 has_topotoolbox <- function(a){
-	return(.C("has_topotoolbox",as.integer(a)))
+	return(.C(C_has_topotoolbox,as.integer(a)))
 
diff --git a/R/topotoolboxr-package.R b/R/topotoolboxr-package.R
index b988824..d688153 100644
--- a/R/topotoolboxr-package.R
+++ b/R/topotoolboxr-package.R
@@ -1,4 +1,4 @@
 ## usethis namespace: start
-#' @useDynLib topotoolboxr, .registration = TRUE
+#' @useDynLib topotoolboxr, .registration = TRUE, .fixes = "C_"
 ## usethis namespace: end
 NULL
```

When you run `check()`, this will regenerate `NAMESPACE` so you will need to commit the changed `NAMESPACE` file again as well.